rllib/algorithms/algorithm.py

```diff
@@ -581,107 +576,50 @@ def step(self) -> ResultDict:
-                                                self.config["evaluation_config"],
-                                                train_future,
-                                            )
-                                        )
-                                    )
-                                # Run `self.evaluate()` only once per iteration.
-                                elif first_step_attempt:
-                                    first_step_attempt = False
-                                    result.update(self.evaluate())
-                                # Collect the training results from the future.
-                                result.update(train_future.result())
-
-                        # Sequential: train (already done above), then eval.
-                        if not self.config["evaluation_parallel_to_training"]:
-                            result.update(self.evaluate())
-
-                    # Collect rollout worker metrics.
-                    episodes, self._episodes_to_be_collected = collect_episodes(
-                        local_worker,
-                        self._remote_workers_for_metrics,
-                        self._episodes_to_be_collected,
-                        timeout_seconds=self.config[
-                            "metrics_episode_collection_timeout_s"
-                        ],
-                    )
-                    self._rollout_worker_metrics.extend(episodes)
-                # @ray.remote RolloutWorker failure.
-                except RayError as e:
-                    # Try to recover w/o the failed worker.
-                    if (
-                        self.config["ignore_worker_failures"]
-                        or self.config["recreate_failed_workers"]
-                    ):
-                        logger.exception("Error in train call, attempting to recover")
-                        self.try_recover_from_step_attempt()
-                    # Error out.
-                    else:
-                        logger.warning(
-                            "Worker crashed during call to `step_attempt()`. "
-                            "To try to continue training without failed "
-                            "worker(s), set `ignore_worker_failures=True`. "
-                            "To try to recover the failed worker(s), set "
-                            "`recreate_failed_workers=True`."
-                        )
-                        raise e
-                # Any other exception.
-                except Exception as e:
-                    # Allow logs messages to propagate.
-                    time.sleep(0.5)
-                    raise e
+        # - No evaluation necessary, just run the next training iteration.
+        # - We have to evaluate in this training iteration, but no parallelism ->
+        #   evaluate after the training iteration is entirely done.
+        if not evaluate_this_iter or not self.config["evaluation_parallel_to_training"]:
```

Comment:
While we're at it I think we could get of the double _if evaluate_this_iter and not self.config["evaluation_parallel_to_training"]_ in L594/L582 ?

Reply:
? 
You mean we should make the if check a positive one? Like
```
if evaluate_this_iter and self.config["evaluation_parallel_to_training"]:
  ...
else
  ...
```